lib/beaker/host.rb

```diff
@@ -389,6 +390,79 @@ def do_scp_from source, target, options
       return result
     end
 
+    # rsync a file or directory from the localhost to this test host
+    # @param from_path [String] The path to the file/dir to upload
+    # @param to_path [String] The destination path on the host
+    # @param opts [Hash{Symbol=>String}] Options to alter execution
+    # @option opts [Array<String>] :ignore An array of file/dir paths that will not be copied to the host
+    def do_rsync_to from_path, to_path, opts = {}
+      ssh_opts = self['ssh']
+      rsync_args = []
+      ssh_args = []
+
+      if not File.file?(from_path) and not File.directory?(from_path)
+        raise IOError, "No such file or directory - #{from_path}"
+      end
+
+      # We enable achieve mode and compression
+      rsync_args << "-az"
+
+      if not ssh_opts.has_key?('user')
+        user = "root"
+      else
+        user = ssh_opts['user']
+      end
+      host = "#{user}@#{self}"
+
+      Rsync.host = host
+
+      if ssh_opts.has_key?('keys') and
+        ssh_opts.has_key?('auth_methods') and
+        ssh_opts['auth_methods'].include?('publickey')
+
+        key = ssh_opts['keys']
+
+        # If an array was set, then we use the first value
+        if key.is_a? Array
+          key = key.first
+        end
+
+        # We need to expand tilde manually as rsync can be
+        # funny sometimes
+        key = File.expand_path(key)
+
+        ssh_args << "-i #{key}"
+      end
+
+      if ssh_opts.has_key?('port') and
+        ssh_args << "-p #{ssh_opts['port']}"
+      end
+
+      # We disable prompt when host isn't known
+      ssh_args << "-o 'StrictHostKeyChecking no'"
+
+      if not ssh_args.empty?
+        rsync_args << "-e \"ssh #{ssh_args.join(' ')}\""
+      end
+
+      if opts.has_key?(:ignore) and not opts[:ignore].empty?
+        opts[:ignore].map! do |value|
+          "--exclude '#{value}'"
```

Comment:
Does 'exclude' behave in the same way as 'ignore', just want to make sure that we don't slightly differ from expected behavior.